swift-style-guide-en.md

```diff
@@ -0,0 +1,1670 @@
+
+```swift
+func pirateName() -> (firstName: String, lastName: String) {
+    return ("Guybrush", "Threepwood")
+}
+
+let name = pirateName()
+let firstName = name.firstName
+let lastName = name.lastName
+```
+
+* **5.1.6** Be wary of retain cycles when creating delegates/protocols for your classes; typically, these properties should be declared `weak`.
+
+* **5.1.7** Be careful when calling `self` directly from an escaping closure as this can cause a retain cycle - use a [capture list](https://developer.apple.com/library/ios/documentation/swift/conceptual/Swift_Programming_Language/Closures.html#//apple_ref/doc/uid/TP40014097-CH11-XID_163) when this might be the case. Extend object lifetime using the `[weak self]` and `guard let strongSelf = self else { return }` idiom. `[weak self]` is preferred to `[unowned self]` where it is not immediately obvious that `self` outlives the closure. Explicitly extending lifetime is preferred to optional unwrapping.
+
+**Preferred**
+```swift
+resource.request().onComplete { [weak self] response in
+    guard let strongSelf = self else { return }
+    let model = strongSelf.updateModel(response)
+    strongSelf.updateUI(model)
+}
+```
+
+**Not Preferred**
+```swift
+// might crash if self is released before response returns
+resource.request().onComplete { [unowned self] response in
+    let model = self.updateModel(response)
+    self.updateUI(model)
+}
+```
+
+**Not Preferred**
+```swift
+// deallocate could happen between updating the model and updating UI
+resource.request().onComplete { [weak self] response in
+    let model = self?.updateModel(response)
+    self?.updateUI(model)
+}
+```
+
+
+* **5.1.8** Don't use labeled breaks.
+
+* **5.1.9** Avoid writing out an `enum` type where possible - use shorthand.
+
+**Preferred:**
+
+```swift
+imageView.setImageWithURL(url, type: .person)
+```
+
+**Not Preferred:**
+
+```swift
+imageView.setImageWithURL(url, type: AsyncImageView.Type.person)
+```
+
+* **5.1.10** Don’t use shorthand for class methods since it is generally more difficult to infer the context from class methods as opposed to `enum`s.
+
+**Preferred:**
+
+```swift
+// .white is not enum
+imageView.backgroundColor = UIColor.white
+```
+
+**Not Preferred:**
+
+```swift
+imageView.backgroundColor = .white
+```
+
+* **5.1.11** Prefer not writing `self.` unless it is required.
+
+* **5.1.12** When writing methods, keep in mind whether the method is intended to be overridden or not. If not, mark it as `final`, though keep in mind that this will prevent the method from being overwritten for testing purposes. In general, `final` methods result in improved compilation times, so it is good to use this when applicable. Be particularly careful, however, when applying the `final` keyword in a library since it is non-trivial to change something to be non-`final` in a library as opposed to have changing something to be non-`final` in your local project.
+
```

Comment:
`final` methods result in improved compilation times, so it is good to use this when applicable